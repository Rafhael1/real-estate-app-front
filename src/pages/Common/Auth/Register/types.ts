/** @format */

export const ACTIONS = {
  CREATE_USER_REQUEST: 'CREATE_USER_REQUEST',
  CREATE_USER_SUCCESS: 'CREATE_USER_SUCCESS',
  CREATE_USER_ERROR: 'CREATE_USER_ERROR',

  LOGIN_REQUEST: 'LOGIN_REQUEST',
  LOGIN_SUCCESS: 'LOGIN_SUCCESS',
  LOGIN_ERROR: 'LOGIN_ERROR',

  IS_LOGGED_REQUEST: 'IS_LOGGED_REQUEST',
  IS_LOGGED_SUCCESS: 'IS_LOGGED_SUCCESS',
  IS_LOGGED_ERROR: 'IS_LOGGED_ERROR'
}

export type UserType = {
  id: string
  name: string
  email: string
  password: string
}

export interface CreateUserRequest {
  type: typeof ACTIONS.CREATE_USER_REQUEST
}

export interface CreateUserSuccess {
  type: typeof ACTIONS.CREATE_USER_SUCCESS
  payload: UserType
}

export interface CreateUserError {
  type: typeof ACTIONS.CREATE_USER_ERROR
}

export type CreateUserDispatchTypes =
  | CreateUserRequest
  | CreateUserSuccess
  | CreateUserError

export interface IsLoggedRequest {
  type: typeof ACTIONS.IS_LOGGED_REQUEST
}

export interface IsLoggedSuccess {
  type: typeof ACTIONS.IS_LOGGED_SUCCESS
}

export interface IsLoggedError {
  type: typeof ACTIONS.IS_LOGGED_ERROR
}

export type IsLoggedDispatchTypes =
  | IsLoggedRequest
  | IsLoggedSuccess
  | IsLoggedError

export interface LoginRequest {
  type: typeof ACTIONS.LOGIN_REQUEST
}

export interface LoginSuccess {
  type: typeof ACTIONS.LOGIN_SUCCESS
}

export interface LoginError {
  type: typeof ACTIONS.LOGIN_ERROR
}

export type LoginDispatchTypes = LoginRequest | LoginSuccess | LoginError
