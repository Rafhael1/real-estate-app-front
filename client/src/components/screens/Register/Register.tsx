import React, { useState } from "react"

import axios from "../../utils/api/axios"

const Register = () => {

	const [name, setName] = useState("")
	const [email, setEmail] = useState("")
	const [password, setPassword] = useState("")


	const onSubmit = async (e: any) => {
		e.preventDefault()

		const body = {
			name,
			email,
			password
		}

		const res = await axios.post("/user/register", {
			headers: {
				"Content-Type": "application/json",
			},
			body: body
		})
		console.log(res)
	}

	return (
		<form onSubmit={(e) => onSubmit(e)}>
			<input placeholder="name" value={name} onChange={(e) => setName(e.target.value)} />
			<input placeholder="email" value={email} onChange={(e) => setEmail(e.target.value)}  />
			<input placeholder="password" value={password} onChange={(e) => setPassword(e.target.value)}  />
			<button type="submit" >Send</button>
		</form>
	)
}

export default Register