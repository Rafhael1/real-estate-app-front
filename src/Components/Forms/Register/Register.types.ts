import { InjectedFormProps } from 'redux-form';

export type RegisterProps = {
  handleSubmit?: InjectedFormProps['handleSubmit'];
  isModalOpen?: boolean;
  handleModalOpen?: any;
};
