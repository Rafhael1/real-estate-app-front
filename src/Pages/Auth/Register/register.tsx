import React from 'react';
import { Form, Field, reduxForm } from 'redux-form';
import { Stack, Button } from '@mui/material';
import { UserType } from './register.types';
import { useAppDispatch } from '../../../Redux/hooks';

import renderTextField from '../../../Components/Fields/inputField';

import { register } from './register.actions';

const Register = ({ handleSubmit }: any) => {
  const dispatch = useAppDispatch();

  return (
    <Form
      onSubmit={handleSubmit((values: UserType) => dispatch(register(values)))}
    >
      <Stack
        direction="column"
        justifyContent="center"
        alignItems="center"
        spacing={0}
      >
        <Field
          name="name"
          variant="outlined"
          color="primary"
          label="Name"
          type="text"
          component={renderTextField}
        />
        <Field
          name="email"
          variant="outlined"
          color="primary"
          label="Email"
          type="email"
          component={renderTextField}
        />
        <Field
          name="password"
          variant="outlined"
          color="primary"
          label="Password"
          type="password"
          component={renderTextField}
        />
        <Field
          component={Button}
          type="submit"
          props={{
            variant: 'contained',
            color: 'primary'
          }}
        >
          Sign Up
        </Field>
      </Stack>
    </Form>
  );
};

export default reduxForm({
  form: 'RegisterForm'
})(Register);
