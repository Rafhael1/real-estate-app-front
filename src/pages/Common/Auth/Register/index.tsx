import React from 'react'
import { Form, Field, reduxForm } from 'redux-form'
import { Button } from '@mui/material'
import { UserType } from './types'
import { useAppDispatch } from '../../../../redux/hooks'

import renderTextField from '../../../../components/fields/inputField'

import { register } from './redux/actions'

const Register = ({ handleSubmit }: any) => {
  const dispatch = useAppDispatch()

  return (
    <Form
      onSubmit={handleSubmit((values: UserType) => dispatch(register(values)))}
    >
      <Field
        name="name"
        variant="outlined"
        color="primary"
        label="Name"
        type="text"
        component={renderTextField}
      />
      <Field
        name="email"
        variant="outlined"
        color="primary"
        label="Email"
        type="email"
        component={renderTextField}
      />
      <Field
        name="password"
        variant="outlined"
        color="primary"
        label="Password"
        type="password"
        component={renderTextField}
      />
      <Field
        component={Button}
        type="submit"
        props={{
          variant: 'contained',
          color: 'primary'
        }}
      >
        Sign Up
      </Field>
    </Form>
  )
}

export default reduxForm({
  form: 'RegisterForm'
})(Register)
