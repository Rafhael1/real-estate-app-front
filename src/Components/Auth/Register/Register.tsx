import React from 'react';
import { Form, Field, reduxForm } from 'redux-form';
import { Stack, Button } from '@mui/material';
import { UserType } from './Register.types';
import { useDispatch } from '../../../Redux/Hooks';

import renderTextField from '../../Fields/InputField';

import { register } from './Register.actions';

const Register = ({ handleSubmit }: any) => {
  const dispatch = useDispatch();

  return (
    <Form
      onSubmit={handleSubmit((values: UserType) => dispatch(register(values)))}
    >
      <Stack
        direction="column"
        justifyContent="center"
        alignItems="center"
        spacing={0}
      >
        <Field
          name="name"
          variant="outlined"
          color="primary"
          label="Name"
          type="text"
          component={renderTextField}
        />
        <Field
          name="email"
          variant="outlined"
          color="primary"
          label="Email"
          type="email"
          component={renderTextField}
        />
        <Field
          name="password"
          variant="outlined"
          color="primary"
          label="Password"
          type="password"
          component={renderTextField}
        />
        <Field
          component={Button}
          type="submit"
          props={{
            variant: 'contained',
            color: 'primary'
          }}
        >
          Sign Up
        </Field>
      </Stack>
    </Form>
  );
};

export default reduxForm({
  form: 'RegisterForm'
})(Register);
