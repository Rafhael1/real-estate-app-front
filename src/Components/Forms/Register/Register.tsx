import React from 'react';
import { useForm } from 'react-hook-form';
import { useDispatch, useSelector } from '../../../Hooks/Redux';
import { Stack, Dialog, DialogTitle, DialogContent } from '@mui/material';
import { UserType, AuthProps } from 'Types/Auth/Auth.types';

import { TextField, CheckboxField } from 'Components';

import { register } from 'Services/Auth/Auth.actions';
import { LoadingButton } from '@mui/lab';

const Register = ({ isModalOpen, handleModalOpen }: AuthProps) => {
  const dispatch = useDispatch();
  const authState = useSelector((state) => state.Auth);
  const { control, handleSubmit } = useForm();

  return (
    <Dialog
      open={isModalOpen}
      onClose={() => handleModalOpen('register')}
      transitionDuration={1}
      PaperProps={{ sx: { borderRadius: '25px', height: '450px' } }}
    >
      <DialogTitle sx={{ alignSelf: 'center' }}>Create Account</DialogTitle>
      <DialogContent>
        <form
          onSubmit={handleSubmit(async (values: UserType) => {
            await dispatch(register(values));
            handleModalOpen('register');
          })}
        >
          <Stack
            direction="column"
            justifyContent="center"
            alignItems="center"
            spacing={0}
          >
            <TextField
              name="name"
              color="primary"
              label="Name"
              type="text"
              control={control}
            />
            <TextField
              required
              name="email"
              color="primary"
              label="Email"
              type="email"
              control={control}
            />
            <TextField
              required
              name="password"
              color="primary"
              label="Password"
              type="password"
              control={control}
            />
            <LoadingButton
              loading={authState.isLoading}
              sx={{ width: '200px' }}
              type="submit"
            >
              Sign Up
            </LoadingButton>
          </Stack>
        </form>
      </DialogContent>
    </Dialog>
  );
};

export default Register;
