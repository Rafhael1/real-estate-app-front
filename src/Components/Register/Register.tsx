import React from 'react';
import { useForm } from 'react-hook-form';
import { useDispatch, useSelector } from '../../Utils/Hooks/Redux';
import { Stack, Dialog, DialogTitle, DialogContent } from '@mui/material';
import { UserType, AuthProps } from 'Types/Auth/Auth.types';

import { TextField } from 'Components';

import { register } from 'Services/Auth/Auth.actions';
import { LoadingButton } from '@mui/lab';
import useMediaQuery from 'Utils/Hooks/useMediaQuery';

const Register = ({ isModalOpen, handleModalOpen }: AuthProps) => {
	const dispatch = useDispatch();
	const authState = useSelector((state) => state.Auth);
	const { control, handleSubmit } = useForm();
	const { isMobile } = useMediaQuery();

	return (
		<Dialog
			open={isModalOpen}
			onClose={() => handleModalOpen('register')}
			transitionDuration={1}
			PaperProps={{ sx: { borderRadius: '25px', height: '450px' } }}
		>
			<DialogTitle sx={{ alignSelf: 'center' }}>Create Account</DialogTitle>
			<DialogContent>
				<form
					onSubmit={handleSubmit(async (values: UserType) => {
						await dispatch(register(values));
						handleModalOpen('register');
					})}
				>
					<Stack
						direction="column"
						justifyContent="center"
						alignItems="center"
						spacing={0}
					>
						<TextField
							name="name"
							color="primary"
							label="Name"
							type="text"
							control={control}
							sx={{ width: isMobile ? '250px' : null }}
						/>
						<TextField
							required
							name="email"
							color="primary"
							label="Email"
							type="email"
							control={control}
							sx={{ width: isMobile ? '250px' : null }}
						/>
						<TextField
							required
							name="password"
							color="primary"
							label="Password"
							type="password"
							control={control}
							sx={{ width: isMobile ? '250px' : null }}
						/>
						<LoadingButton
							loading={authState.isLoading}
							sx={{ width: '200px' }}
							type="submit"
						>
							Sign Up
						</LoadingButton>
					</Stack>
				</form>
			</DialogContent>
		</Dialog>
	);
};

export default Register;
