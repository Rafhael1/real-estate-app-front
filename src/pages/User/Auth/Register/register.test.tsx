import React from 'react'
import { shallow } from 'enzyme'
import Register from '.'

it('Expect', () => {
  expect(shallow(<Register />)).toMatchSnapshot()
})