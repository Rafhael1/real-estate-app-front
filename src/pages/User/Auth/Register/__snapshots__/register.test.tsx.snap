// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect 1`] = `
<Hoc
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="RegisterForm"
  forwardedRef={null}
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  submitAsSideEffect={false}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
